src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    add(val) {
+      this.result += val;
+
+      return this;
+    },
+    subtract(val) {
+      this.result -= val;
+
+      return this;
+    },
+    multiply(val) {
+      this.result *= val;
+
+      return this;
+    },
+    divide(val) {
+      if (val === 0) {
+        return 'Cannot divide by zero.';
+      } else {
+        this.result /= val;
+      }
+
+      return this;
+    },
+    operate(operation, value) {
+      if (typeof operation === 'function') {
+        operation.call(this, value);
```

Comment:
The `operate` method calls the provided function with `this` as the context, which is correct. However, ensure that the function passed as `operation` is designed to work with this context and the `value` parameter.